vendor/github.com/raintank/schema/metric.go

```diff
@@ -81,6 +85,23 @@ func (m *MetricData) KeyBySeries(b []byte) []byte {
 	return b
 }
 
+func (m *MetricData) KeyBySeriesWithTags(b []byte) []byte {
+	nameWithTagsBuffer := bytes.NewBuffer(b[:0])
```

Comment:
it shouldn't be up to this function to modify pre-existing data in the slice.
if you want to discard previous data (we generally do), it's up to the caller to reset len back to 0.
this function should just append data behind pre-existing data, like the other functions (above) do.

for some examples of this pratice in the standard library, see
* hash.Sum (https://golang.org/pkg/hash/#Hash)
* https://godoc.org/time#Time.AppendFormat

Reply:
i just checked all the call sites, they already honor this pattern (by just passing in nil slices)